src/ImageSharp.Web/Synchronization/RefCountedConcurrentDictionary.cs

```diff
@@ -0,0 +1,243 @@
+                    {
+                        return;
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Get an enumeration over the contents of the dictionary for testing/debugging purposes
+        /// </summary>
+        internal IEnumerable<(TKey Key, TValue Value, int RefCount)> DebugGetContents() => new RefCountedDictionaryEnumerable(this);
+
+        /// <summary>
+        /// Internal class used for testing/debugging purposes
+        /// </summary>
+        private class RefCountedDictionaryEnumerable : IEnumerable<(TKey Key, TValue Value, int RefCount)>
+        {
+            private readonly RefCountedConcurrentDictionary<TKey, TValue> dictionary;
+
+            internal RefCountedDictionaryEnumerable(RefCountedConcurrentDictionary<TKey, TValue> dictionary)
+                => this.dictionary = dictionary;
+
+            public IEnumerator<(TKey Key, TValue Value, int RefCount)> GetEnumerator()
+                => new RefCountedDictionaryEnumerator(this.dictionary);
+
+            IEnumerator IEnumerable.GetEnumerator() => this.GetEnumerator();
+        }
+
+        /// <summary>
+        /// Internal class used for testing/debugging purposes
+        /// </summary>
+        private class RefCountedDictionaryEnumerator : IEnumerator<(TKey Key, TValue Value, int RefCount)>
+        {
+            private readonly IEnumerator<KeyValuePair<TKey, RefCountedValue>> enumerator;
+
+            public RefCountedDictionaryEnumerator(RefCountedConcurrentDictionary<TKey, TValue> dictionary)
+                => this.enumerator = dictionary.dictionary.GetEnumerator();
+
+            public (TKey Key, TValue Value, int RefCount) Current
+            {
+                get
+                {
+                    KeyValuePair<TKey, RefCountedValue> keyValuePair = this.enumerator.Current;
+                    return (keyValuePair.Key, keyValuePair.Value.Value, keyValuePair.Value.RefCount);
+                }
+            }
+
+            object IEnumerator.Current => this.Current;
+
+            public void Dispose() => this.enumerator.Dispose();
+
+            public bool MoveNext() => this.enumerator.MoveNext();
+
+            public void Reset() => this.enumerator.Reset();
+        }
+
+        /// <summary>
+        /// Simple immutable tuple that combines a <typeparamref name="TValue"/> instance with a ref count integer.
+        /// </summary>
+        private class RefCountedValue : IEquatable<RefCountedValue>
+        {
+#pragma warning disable SA1401 // Fields should be private
+            public readonly TValue Value;
+            public readonly int RefCount;
+#pragma warning restore SA1401 // Fields should be private
+
+            public RefCountedValue(TValue value, int refCount)
+            {
+                this.Value = value;
+                this.RefCount = refCount;
+            }
+
+            public bool Equals(
```

Comment:
I am getting ` error CS8767: Nullability of reference types in type of parameter 'other' of 'bool RefCountedValue.Equals(RefCountedValue other)' doesn't match implicitly implemented member 'bool IEquatable<RefCountedValue>.Equals(RefCountedValue other)' (possibly because of nullability attributes)` when running the benchmarks

Reply:
Weird.  This compiles cleanly in both netcoreapp2.1 and 3.1 for me, and I'm able to run the benchmarks in both TFMs as well....?  @sebastienros Anything special about how you're trying to run the benchmarks?

Reply:
To be more precise, using crank, doing a web load benchmark independent from BDN. It works fine on master though. Full stack:

```
Command:
dotnet publish ImageSharp.Web.Sample.csproj -c Release -o /tmp/benchmarks-agent/benchmarks-server-1/2ns2okc1.ov0/ImageSharp.Web/samples/ImageSharp.Web.Sample/published /p:MicrosoftNETCoreAppPackageVersion=3.1.21 /p:MicrosoftAspNetCoreAppPackageVersion=3.1.21 /p:MicrosoftNETCoreApp31PackageVersion=3.1.21 /p:MicrosoftNETPlatformLibrary=Microsoft.NETCore.App /p:RestoreNoCache=true --framework netcoreapp3.1 --self-contained -r linux-x64 
Microsoft (R) Build Engine version 16.7.2+b60ddb6f4 for .NET
Copyright (C) Microsoft Corporation. All rights reserved.

  Determining projects to restore...
  Restored /tmp/benchmarks-agent/benchmarks-server-1/2ns2okc1.ov0/ImageSharp.Web/src/ImageSharp.Web/ImageSharp.Web.csproj (in 297 ms).
  Restored /tmp/benchmarks-agent/benchmarks-server-1/2ns2okc1.ov0/ImageSharp.Web/samples/ImageSharp.Web.Sample/ImageSharp.Web.Sample.csproj (in 297 ms).
Synchronization/RefCountedConcurrentDictionary.cs(229,25): error CS8767: Nullability of reference types in type of parameter 'other' of 'bool RefCountedValue.Equals(RefCountedValue other)' doesn't match implicitly implemented member 'bool IEquatable<RefCountedValue>.Equals(RefCountedValue other)' (possibly because of nullability attributes). [/tmp/benchmarks-agent/benchmarks-server-1/2ns2okc1.ov0/ImageSharp.Web/src/ImageSharp.Web/ImageSharp.Web.csproj]
Exit code: 1
```

Reply:
I just pushed a small change to this code that adjusts the nullability attributes when compiling with net5.0 or higher.  This is preemptively getting ahead of any update to this codebase to target net6.0, and _maybe_ it will address whatever issue you hit running benchmarks?

Reply:
With your changes it builds on net6.0. At least it's unblocking me.